### PR TITLE
feat: implement async rest lro methods

### PR DESCRIPTION
**Pull request by parthea**
This PR adds support for LROs and IAM and Locations mixins.
Fixes: https://github.com/googleapis/gapic-generator-python/issues/2170

### CODE REVIEW

**Review by vchudnov-g — APPROVED** [review]
Looks good! Please do add a PR description summarizing the changes.

File: gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2 [hunk 1]
```diff
@@ -20,6 +21,16 @@ from google.auth.aio import credentials as ga_credentials_async  # type: ignore
 
 from google.api_core import exceptions as core_exceptions
 from google.api_core import gapic_v1
+{% if service.has_lro %}
+from google.api_core import operations_v1
+{% endif %}{# service.has_lro #}
+{% if opts.add_iam_methods or api.has_iam_mixin %}
```
[the remark this answers is not in the document]
Comment (ohmayr) [reply gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest_asyncio.py.j2:27]: addressed.

File: gapic/templates/setup.py.j2 [hunk 2]
```diff
@@ -51,6 +51,7 @@ dependencies = [
 extras = {
 {% if rest_async_io_enabled %}
     "async_rest": [
+        {# TODO(https://github.com/googleapis/python-api-core/pull/724): Update the minimum supported version of api-core once the linked PR is merged and released. #}
```
[the remark this answers is not in the document]
Comment (ohmayr) [reply gapic/templates/setup.py.j2]: addressed.

File: gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2 [hunk 3]
```diff
@@ -1771,6 +1565,31 @@ def test_transport_kind_{{ transport_name }}():
 
 {% endmacro %}{# transport_kind_test #}
 
+{% macro lro_client_test(service, api, transport, is_async) %}
+{% if 'rest' in transport %}
+{% set transport_name = get_transport_name(transport, is_async) %}
+def test_{{ service.name|snake_case }}_{{ transport_name }}_lro_client():
```
[the remark this answers is not in the document]
Comment (ohmayr) [reply gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2:1571]: The removed test cases are now moved here within this macro.

**Pull request merged**